dome9/data_source_dome9_admission_control_policy_test.go

```diff
@@ -0,0 +1,55 @@
+package dome9
+
+import (
+	"github.com/terraform-providers/terraform-provider-dome9/dome9/common/testing/variable"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+
+	"github.com/terraform-providers/terraform-provider-dome9/dome9/common/resourcetype"
+	"github.com/terraform-providers/terraform-provider-dome9/dome9/common/testing/method"
+)
+
+func TestAccDataSourceAdmissionControlPolicyBasic(t *testing.T) {
+	// Generate All Required Random Names for Testing
+	policyTypeAndName, policyDataSourceTypeAndName, policyGeneratedName := method.GenerateRandomSourcesTypeAndName(resourcetype.AdmissionControlPolicy)
+	notificationTypeAndName, _, notificationGeneratedName := method.GenerateRandomSourcesTypeAndName(resourcetype.ContinuousComplianceNotification)
+	kubernetesAccountResourceTypeAndName, _, kubernetesAccountGeneratedName := method.GenerateRandomSourcesTypeAndName(resourcetype.CloudAccountKubernetes)
+
+	// Create Kubernetes Account HCL Resource
+	kubernetesAccountHCL := getCloudAccountKubernetesResourceHCLWithfeatures(kubernetesAccountGeneratedName, variable.AdmissionControlKubernetesAccountName,
+		variable.CloudAccountKubernetesRuntimeProtectionEnabled,
+		variable.CloudAccountKubernetesAdmissionControlEnabled,
+		variable.CloudAccountKubernetesImageAssuranceEnabled)
+
+	// Generate Notification HCL Configurations
+	notificationHCL := getContinuousComplianceNotificationResourceHCL(notificationGeneratedName, continuousComplianceNotificationConfig())
+
+	// Create Admission Control Policy HCL Resource
+	admissionPolicyHCL := getAdmissionControlPolicyResourceHCL(kubernetesAccountHCL, kubernetesAccountResourceTypeAndName, notificationHCL,
+		notificationTypeAndName, policyGeneratedName, false)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAdmissionControlPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAdmissionControlPolicyBasic(admissionPolicyHCL, policyGeneratedName, policyTypeAndName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(policyDataSourceTypeAndName, "action", policyTypeAndName, "action"),
+					resource.TestCheckResourceAttrPair(policyDataSourceTypeAndName, "id", policyTypeAndName, "id"),
+					resource.TestCheckResourceAttrPair(policyDataSourceTypeAndName, "target_id", policyTypeAndName, "target_id"),
+					resource.TestCheckResourceAttr(policyTypeAndName, "notification_ids.#", "2"),
+					resource.TestCheckResourceAttrSet(policyTypeAndName, "notification_ids.0"),
+					resource.TestCheckResourceAttrSet(policyTypeAndName, "notification_ids.1"),
+					resource.TestCheckResourceAttrPair(policyDataSourceTypeAndName, "ruleset_id", policyTypeAndName, "ruleset_id"),
+					resource.TestCheckResourceAttrPair(policyDataSourceTypeAndName, "target_type", policyTypeAndName, "target_type"),
+				),
+				ExpectNonEmptyPlan: true,
```

Comment:
I remember this field is needed while we testing the terraform apply in my test scenario, what happened here the test generate HCL Resources : 

1. Create Kubernetes Cloud Account and Enable on it AC and others features
2. Create Compliance Notification
3. Create AC Policy

At step one which occurrs first, the default policy for the newly created account will be created, then the terrafom comes to step 3, and when terrafom core runs apply from go, it detect that there will be a diff on the resource ac policy, so it identifies there is a change there and failing the plan diffs check, so to tell terrafom go sdk to expect there is a diff while running plan, we need this flag to be set, it happens only with our flow since we created and updated the resource as explained

Reply:
if i remove it, tf plan will identity there is a change between what tf apply try to do and what is the current state on the dome9, which is IMO is the esxpected behaviour and test flow, we can see here the test failed if we renmove this important flag in our scenario : 
== RUN   TestAccResourceAdmissionControlPolicyBasic
    testing.go:569: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: data.dome9_admission_control_policy.csufpbiaol
          action:           "" => "<computed>"
          id:               "" => "3737b2a9-5fa8-4403-a98e-d150a35d8cb7"
          notification_ids: "" => "<computed>"
          ruleset_id:       "" => "<computed>"
          target_id:        "" => "<computed>"
          target_type:      "" => "<computed>"
        UPDATE: dome9_admission_control_policy.csufpbiaol
          action:             "Detection" => "Detection"
          id:                 "3737b2a9-5fa8-4403-a98e-d150a35d8cb7" => "3737b2a9-5fa8-4403-a98e-d150a35d8cb7"
          notification_ids.#: "2" => "1"
          notification_ids.0: "d529923f-714e-4916-8d5c-ba88aa2ab97d" => "d529923f-714e-4916-8d5c-ba88aa2ab97d"
          notification_ids.1: "52008dbc-6fdc-45d5-b588-de2afe13ac5d" => ""
          ruleset_id:         "-2001" => "-2001"
          target_id:          "b5f390e0-9e31-40dc-a966-601bd2031ce3" => "b5f390e0-9e31-40dc-a966-601bd2031ce3"
          target_type:        "Environment" => "Environment"